sda-commons-server-spring-data-mongo/src/main/java/org/sdase/commons/server/spring/data/mongo/SpringDataMongoBundle.java

```diff
@@ -134,12 +144,41 @@ public void run(C configuration, Environment environment) {
 
     var cs = new ConnectionString(connectionString);
     this.database = cs.getDatabase();
-    mongoClient = MongoClients.create(cs);
+
+    mongoClient = createMongoClient(cs);
 
     registerHealthCheck(environment.healthChecks(), mongoClient, this.database);
     registerOnShutdown(environment);
   }
 
+  public MongoClient createMongoClient(ConnectionString cs) {
```

Comment:
Imho this could be private. Public access is available by `#mongoClient()`

```suggestion
  private MongoClient createMongoClient(ConnectionString cs) {
```

Reply:
You are right